pkg/util/cache/cache.go

```diff
@@ -27,6 +27,9 @@ const (
 type Cache []*cacheShard
 
 func NewCache(maxSize int) Cache {
+	if maxSize < shardsCount {
```

Comment:
@lavalamp - regarding why this cache exists. It was added before 1.0 release, where we didn't have "cacher" in apiserver, and thus we were unnecessary deserializing the same object multiple times (as it was send to different watchers). It's possible that currently it doesn't provide much value and we can remove it. We should evaluate it, but I would wait for protobufs to be in, to really confirm if it is useful (I guess it won't be at that point).